app/controllers/frontend_log_controller.rb

```diff
@@ -5,9 +5,20 @@ class FrontendLogController < ApplicationController
   before_action :check_user_authenticated
   before_action :validate_parameter_types
 
+  EVENT_MAP = {
+    'IdV: personal key visited' => :idv_personal_key_visited,
+    'IdV: personal key confirm visited' => :idv_personal_key_confirm_visited,
+    'IdV: personal key submitted' => :idv_personal_key_submitted,
+  }.freeze
+
   def create
-    event = "Frontend: #{log_params[:event]}"
-    analytics.track_event(event, log_params[:payload].to_h)
+    event = log_params[:event]
+    payload = log_params[:payload].to_h
+    if EVENT_MAP.key?(event)
+      analytics.public_send(EVENT_MAP[event], **payload)
```

Comment:
I notice `send` seems to get a fair bit of review feedback. An alternative I'm open to here is something more like...


```rb
EVENT_MAP = {
    'IdV: personal key visited' => ->(analytics, **kwargs) { analytics.idv_personal_key_visited(**kwargs) }
}

# ...

if EVENT_MAP.key?(event)
  EVENT_MAP[event].call(analytics, **payload)
end
```

Reply:
The block form is neat, it gave me another idea:

```ruby
EVENT_MAP = {
    'IdV: personal key visited' => AnalyticsEvents.instance_method(:idv_personal_key_visited)
}

# ...

if EVENT_MAP.key?(event)
  EVENT_MAP[event].bind(analytics).call(**payload)
end
```
The advantage to this one is it's a little terser, and `AnalyticsEvents.instance_method(...)` will fail early if the method is typo'd/not a real method

Reply:
Ooh, I do like that! I had tinkered a bit with some auto-binding of `analytics` which looked something more like `instance_eval(payload, &EVENT_MAP[event])`, but I think yours is a bit less magical (in a good way).

Reply:
Pushed up a version more like your suggestion in [6b56e97](https://github.com/18F/identity-idp/pull/6285/commits/6b56e97364aa116a1565bda7f62b822779f0524a). Kept a bit of magic with `Hash#transform_values` to avoid repetition of `AnalyticsEvents.instance_method(...)`. Also, Rubocop seems to prefer `#bind_call` over `#bind#call`.

Reply:
neat! TIL `bind_call`, and ya the `transform_values` is good because it will still catch typos fairly early